zanata-war/src/main/java/org/zanata/service/impl/TranslationFileServiceImpl.java

```diff
@@ -316,23 +332,23 @@ public FileFormatAdapter getAdapterFor(DocumentType type) {
         }
     }
 
-    @Override
-    public String extractExtension(String fileNameOrExtension) {
-        if (fileNameOrExtension == null || fileNameOrExtension.length() == 0
-                || fileNameOrExtension.endsWith(".")) {
-            // could throw exception here
-            return null;
-        }
-
-        String extension;
-        if (fileNameOrExtension.contains(".")) {
-            extension =
-                    fileNameOrExtension.substring(fileNameOrExtension
-                            .lastIndexOf('.') + 1);
-        } else {
-            extension = fileNameOrExtension;
+    /**
+     * Return adapter for given documentType if present, otherwise return adapter
+     * with given fileName.
+     *
+     * @param documentType
+     * @param fileName
+     * @return
```

Comment:
Move the summary text here, since it is describing the return value. The summary text could be "Get an appropriate adapter for a document type or file name."